prereq_flowchart/majorScrapper.py

```diff
@@ -0,0 +1,109 @@
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+import json
+import re
+import time
+
+class Course:
+    credits: int
+    title:str
+    catalogTitle:str
+    description:str
+    inOr:bool #This is to indicate that the course has options
+    required:bool
+    def __init__(self,credits,title,catalogTitle,description,inOr,required):
+        self.credits=credits
+        self.title=title
+        self.catalogTitle=catalogTitle
+        self.description=description
+        self.inOr=inOr
+        self.required=required
+    def getJSON(self):
+        return{  "credits"      :self.credits,
+                 "title"        :self.title,
+                 "catalogTitle" :self.catalogTitle,
+                 "description"  :self.description,
+                 "inOr"         :self.inOr,
+                 "required"     :self.required}
+
+
+class Major:
+    courses:[]
+    minCredits:int
+    name:str
+    def __init__(self,courses,minCredits,name):
+        self.courses=courses
+        self.minCredits=minCredits
+        self.name=name
+    def getJSON(self):
+        mappedCourses=list(map(lambda c:c.getJSON(),self.courses))
+        return{"courses"   :mappedCourses,
+               "minCredits":self.minCredits,
+               "name"      :self.name}
+
+def getMajorLinks():
+    MAJOR_URL="https://onestop2.umn.edu/pcas/viewCatalogSearchResults.do?campusId=UMNTC&userDefinedSearch=true&keyworjds=&careerId=UGRD&programType=BACC"
+
+    browser=webdriver.Firefox()
+    browser.get(MAJOR_URL)
+    browser.maximize_window()
+
+    numberOfMajors=re.match("[0-9]*",browser.find_element(By.CLASS_NAME,"requirementshead").get_attribute("textContent")).group(0)
+    majorLinks=browser.find_element(By.ID,"appTable").find_element(By.TAG_NAME,"tbody").find_elements(By.CLASS_NAME, "bold100")
+    #Links to different majors
+
+    majorJSONlinks=[]
+    for i in range(int(numberOfMajors)):
+        curMajor=majorLinks[i].find_element(By.TAG_NAME,"a").get_attribute('href')
+        majorJSONlinks.append(curMajor)
+
+    file=open("majorLinks.json","w")
+    file.write(json.dumps(majorJSONlinks))
+    file.close()
+    browser.close()
+
+def scrapeMajors():
+    browser=webdriver.Firefox()
+    browser.maximize_window()
+
+    try:
+        linkFile=open("majorLinks.json","r")
+    except IOError:
+        getMajorLinks()
```

Comment:
would it be better to do this memoization inside `getMajorLinks()`? then it can return a dict instead of having this weird interface where it makes a file, and then the calling code reads the file